enumerable.rb

```diff
@@ -0,0 +1,106 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_count
+    return length unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = 0
+    i = 0
+    while i < arr.size
+      count += 1 if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = []
+    i = 0
+    while i < arr.size
+      count << arr[i] if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_all?
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      return false unless yield(arr[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?(con = false)
+    if block_given?
+      size.times do |item|
+        result = yield(self[item])
+        (return true) if result
+      end
+    elsif con
+      size.times do |item|
+        return true if con == self[item]
+      end
+    else
+      size.times do |item|
+        return true if self[item]
+      end
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79677078-5fce4b00-81aa-11ea-9022-fc1a96b3a63c.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/o/) == %w[dog door rod blade].any?(/o/) => true
```
- Yours retrieves the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79677087-8ab89f00-81aa-11ea-9a7a-639c3a50e773.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/79677090-9c01ab80-81aa-11ea-80b3-2edc8464941a.png)

Reply:
Fixed!